libs/libmilter/Makefile

```diff
@@ -0,0 +1,78 @@
+#
+# Copyright (C) 2017 Val Kulkov <val.kulkov@gmail.com>
+#
+# This is free software, licensed under the MIT License
+#
+#
+
+include $(TOPDIR)/rules.mk
+
+PKG_NAME:=sendmail
+PKG_VERSION:=8.15.2
+PKG_RELEASE:=1
+
+PKG_SOURCE:=$(PKG_NAME).$(PKG_VERSION).tar.gz
+PKG_SOURCE_URL:=http://artfiles.org/sendmail.org/pub/sendmail/ ftp://ftp.cs.berkeley.edu/ucb/sendmail/ http://www.netgull.com/sendmail/ http://sendmail.cybermirror.org/
+PKG_HASH:=24f94b5fd76705f15897a78932a5f2439a32b1a2fdc35769bb1a5f5d9b4db439
+PKG_MAINTAINER:=Val Kulkov <val.kulkov@gmail.com>
+
+PKG_LICENSE:=Sendmail
+PKG_LICENSE_FILES:=LICENSE
+
+PKG_INSTALL:=1
+
+include $(INCLUDE_DIR)/package.mk
+
+MAKE_PATH:=libmilter
+SM_SITECONFIG:=devtools/Site/site.config.m4
+export MILTER_SOVER=1.0.1
+
+define Package/libmilter
+  SECTION:=libs
+  CATEGORY:=Libraries
+  DEPENDS:=+libpthread
+  TITLE:=The sendmail Mail Filter API (Milter) library
+  URL:=http://sendmail.org/
+endef
+
+define Package/libmilter/description
+  The sendmail Mail Filter API (Milter) is designed to allow third-party
+programs access to mail messages as they are being processed in order to
+filter meta-information and content.
+endef
+
+TARGET_CFLAGS += $(FPIC)
+
+define Build/Prepare
+	$(Build/Prepare/Default)
+	mkdir -p $(PKG_INSTALL_DIR)/usr/lib
+	$(CP) files/sharedlibrary.m4 $(PKG_BUILD_DIR)/devtools/M4/UNIX/
+	$(SED) \
+		's,^bldPRODUCT_START(`library'\'',bldPRODUCT_START(`sharedlibrary'\'',g' \
+		$(PKG_BUILD_DIR)/libmilter/Makefile.m4
+	echo 'dnl' > $(PKG_BUILD_DIR)/$(SM_SITECONFIG)
+	echo "define(\`confCC', \`"$(TARGET_CC)"')" >> $(PKG_BUILD_DIR)/$(SM_SITECONFIG)
+	echo "define(\`confLDOPTS', \`"$(TARGET_LDFLAGS)"')" >> $(PKG_BUILD_DIR)/$(SM_SITECONFIG)
+	echo "APPENDDEF(\`confDONT_INSTALL_CATMAN')" >> $(PKG_BUILD_DIR)/$(SM_SITECONFIG)
+	echo "APPENDDEF(\`confNO_MAN_BUILD')" >> $(PKG_BUILD_DIR)/$(SM_SITECONFIG)
+	echo "APPENDDEF(\`confINCOWN', \`"$(USER)"')" >> $(PKG_BUILD_DIR)/$(SM_SITECONFIG)
+	echo "APPENDDEF(\`confINCGRP', \`"$(USER)"')" >> $(PKG_BUILD_DIR)/$(SM_SITECONFIG)
+	echo "APPENDDEF(\`confLIBOWN', \`"$(USER)"')" >> $(PKG_BUILD_DIR)/$(SM_SITECONFIG)
+	echo "APPENDDEF(\`confLIBGRP', \`"$(USER)"')" >> $(PKG_BUILD_DIR)/$(SM_SITECONFIG)
```

Comment:
This block of code changes root ownership of certain destination executables, which is the default config option, to the current user by modifying package configuration so that PKG_INSTALL can be performed under a non-root account. The alternative would be to create a patch to change root ownership to the current user in the package installation script. IMO the first option is more future-proof than the second option. Please advise if I am mistaken about this, and I will introduce a patch to the installation script instead of modifying the config.